src/core-services/shop/mutations/createShop.js

```diff
@@ -83,18 +83,16 @@ export default async function createShop(context, input) {
   const {
     accountId,
     appEvents,
+    checkPermissions,
     collections,
     rootUrl,
     simpleSchemas: {
       Shop: ShopSchema
     },
-    userHasPermission,
     userId
   } = context;
 
-  if (!userHasPermission(["owner", "shop/create"], "__global_roles__")) {
-    throw new ReactionError("access-denied", "User does not have permission");
-  }
+  await checkPermissions(["owner", "shop/create"], "__global_roles__");
```

Comment:
I think you could actually just remove `"__global_roles__"` (pass no second arg) and it would default to that. That will make it easier to move to a different permission plugin.